### ISSUE TITLE
not possible to update object – either InvalidPrimaryKeyAttributesUpdateError or attribute value could not be resolved

### ISSUE THREAD

**Issue by silberistgold** [typedorm/typedorm#167]
Hi there,

we already discussed this in #152 but I think the issue is still there so I created this new issue. See my last comment:

Indeed the test is working. But consider the following:

```
test('allows updating key-only attributes', async () => {
  dcMock.update.mockReturnValue({promise: () => ({})});

  const updateResponse = await entityManager.update(
    TestEntity,
    {
      id: '1',
      tenant: 'NEW_TENANT',
    },
    {
      id: '2',
      tenant: 'NEW_TENANT',
    }
  );
  expect(updateResponse).toBeDefined();
});
```

Error:

`"status" was referenced in USER#TENANT#{{tenant}}#STATUS#{{status}} but it's value could not be resolved.`

If `status` is added to the body the error is:

```
InvalidPrimaryKeyAttributesUpdateError: Primary key attribute and non primary key attributes can not be
        updated together. When updating attributes that references a primary key, it can not have
        additional non key attributes.
        
     Affected Primary key attributes: {"PK":"USER#2#TENANT#NEW_TENANT","SK":"USER#2#TENANT#NEW_TENANT"}
        
     Tried to update non-key attributes: ["status"]
```

So as I see it updating non-key attributes works, updating key attributes only does not work (bug) and updating non-key and key attributes together does not work (expected temporarily).

_Originally posted by @silberistgold in https://github.com/typedorm/typedorm/issues/152#issuecomment-900972304_

**Comment by whimzyLive** [comment]
Hi @silberistgold, 
Updating non-key and primary key attributes should be possible once again after #153 is merged.

This will also unblock updating key-only attributes. Therefore,  this will be allowed:
e.g.

```typescript
const updateResponse = await entityManager.update(
    TestEntity,
    {
      id: '1',
      tenant: 'NEW_TENANT',
    },
    {
      id: '2',
      tenant: 'NEW_TENANT',
      status: true
    }
  );
```

**Comment by silberistgold** [comment]
Great. Thanks a lot.

**Comment by whimzyLive** [comment]
Closing as this is related to #153
